spec/classes/hiera_spec.rb

```diff
@@ -528,5 +528,117 @@
         it { is_expected.to contain_class('hiera::deep_merge') }
       end
     end
+    context 'hiera version 5' do
+      let (:facts) do
+        {
+          puppetversion: Puppet.version,
```

Comment:
can you migrate this to rspec-puppet-facts? An example is available at: https://github.com/voxpupuli/puppet-zabbix/blob/master/spec/classes/database_mysql_spec.rb#L8-L12

Feel free to join our IRC channel #voxpupuli on freenode if you've any questions.